ydb/core/ydb_convert/table_description.cpp

```diff
@@ -802,38 +801,53 @@ void FillPartitioningSettings(TYdbProto& out, const NKikimrSchemeOp::TPartitioni
     }
 }
 
+template <typename TYdbProto>
+void FillPartitioningSettingsImpl(TYdbProto& out,
+        const NKikimrSchemeOp::TTableDescription& in) {
+
+    auto& outPartSettings = *out.mutable_partitioning_settings();
+
+    if (!in.HasPartitionConfig()) {
+        FillDefaultPartitioningSettings(outPartSettings);
+        return;
+    }
+
+    const auto& partConfig = in.GetPartitionConfig();
+    if (!partConfig.HasPartitioningPolicy()) {
+        FillDefaultPartitioningSettings(outPartSettings);
+        return;
+    }
+
+    FillPartitioningSettings(outPartSettings, partConfig.GetPartitioningPolicy());
+}
+
 void FillGlobalIndexSettings(Ydb::Table::GlobalIndexSettings& settings,
-        const NKikimrSchemeOp::TIndexDescription& tableIndex,
-        const NKikimrMiniKQL::TType& splitKeyType) {
+    const google::protobuf::RepeatedPtrField<NKikimrSchemeOp::TTableDescription>& indexImplTables
+) {
+    if (indexImplTables.empty()) {
+        return;
+    }
+    const auto& indexImplTableDescription = indexImplTables.Get(0);
 
-    switch (tableIndex.GetPartitionsCase()) {
-    case NKikimrSchemeOp::TIndexDescription::kUniformPartitions:
-        settings.set_uniform_partitions(tableIndex.GetUniformPartitions());
-        break;
-    case NKikimrSchemeOp::TIndexDescription::kExplicitPartitions:
-        FillTableBoundaryImpl(*settings.mutable_partition_at_keys(),
-            tableIndex.GetExplicitPartitions().GetSplitBoundary(),
+    if (indexImplTableDescription.HasUniformPartitionsCount()) {
+        settings.set_uniform_partitions(indexImplTableDescription.GetUniformPartitionsCount());
+    }
+    if (indexImplTableDescription.SplitBoundarySize()) {
```

Comment:
You could ask yourself: "What if __BOTH__ UniformPartitionsCount and SplitBoundary are specified?" My answer is: SplitBoundary has higher priority. The other option I see here is to throw an error. (FillIndexDescription already throws if FillColumnDescription cannot decide how to fill the columns.)

Reply:
> The other option I see here is to throw an error. (FillIndexDescription already throws if FillColumnDescription cannot decide how to fill the columns.)

Exception sounds better for me, because 
1. looks like error case
2. can be changed to something else in future

Reply:
I checked how DescribeTable deals with this problem and found out that it just ignores UniformPartitionsCount, i.e. it does not even check if it exists in the TTableDescription. See the code around [here](https://github.com/ydb-platform/ydb/blob/29e1e4546446acf94c2a350d97b3dbfdf68f766f/ydb/core/tx/schemeshard/schemeshard_path_describer.cpp#L1183). The motivation is: there is no such thing as UniformPartitionsCount for an __existing__ table. If a table already exists, then it has __explicit__ partition boundaries which are represented by SplitBoundary field of the TTableDescription protobuf.